parquet-avro/src/main/java/org/apache/parquet/avro/AvroWriteSupport.java

```diff
@@ -423,7 +435,7 @@ private static GenericData getDataModel(Configuration conf, Schema schema) {
 
     Class<? extends AvroDataSupplier> suppClass = conf.getClass(
         AVRO_DATA_SUPPLIER, SpecificDataSupplier.class, AvroDataSupplier.class);
-    return ReflectionUtils.newInstance(suppClass, conf).get();
+    return ReflectionUtils.newInstance(suppClass, ConfigurationUtil.createHadoopConfiguration(conf)).get();
```

Comment:
Does it mean that we still cannot get rid of the hadoop dependency? Anyway we have to depend on Configuration class here.

Reply:
For now this seems necessary to me in order not to introduce any breaking changes. AFAIK clients _could_ make suppClass be their own implementation, which under Parquet prior to this change would involve invoking setConf(Configuration) if that class were Hadoop Configurable (which is what Hadoop's ReflectionUtils.newInstance(Class, Configuration) also does under the hood).